utbot-framework/src/main/kotlin/org/utbot/framework/plugin/api/TestCaseGenerator.kt

```diff
@@ -116,9 +121,72 @@ open class TestCaseGenerator(
         if (UtSettings.testMinimizationStrategyType == TestSelectionStrategyType.DO_NOT_MINIMIZE_STRATEGY) {
             executions
         } else {
-            minimizeTestCase(executions) { it.result::class.java }
+            minimizeTestCase(executions.filterCoveredInstructions()) { it.result::class.java }
         }
 
+    private fun List<UtExecution>.filterCoveredInstructions(): List<UtExecution> {
+        // Do nothing when we launched not from IDEA
+        if (buildDirs.isEmpty()) return this
+
+        val entityAnnotation = utContext.classLoader.tryLoadClass("javax.persistence.Entity")
+
+        val urls = buildDirs.map { it.toURL() }.toTypedArray()
+        val classLoader = URLClassLoader(urls)
+        val loadedClasses = mutableMapOf<String, Class<*>?>()
+
+        // Here we filter instructions from third-party libraries.
+        // Also, we filter out instructions that operate in classes marked as @Entity.
+        return this.map { execution ->
+
+            fun List<Instruction>.filter(): List<Instruction> {
+                fun ClassLoader.load(fqName: String): Class<*>? =
+                    loadedClasses.getOrPut(fqName) {
+                        this.tryLoadClass(fqName)
+                    }
+
+                return this.mapNotNull { instruction ->
```

Comment:
There's not mapping, just filtering
```suggestion
                return this.filter { instruction ->
```